EIPS/eip-4786.md

```diff
@@ -0,0 +1,397 @@
+    ) external;
+
+    /**
+     * @dev unlink ERC-1155 token to `to`
+     */
+    function unlinkERC1155(
+        address to,
+        ERC1155Token memory erc1155Token,
+        uint256 amount,
+        ERC721Token memory targetToken,
+        bytes memory data
+    ) external;
+}
+```
+
+
+
+## Rationale
+
+The contract records the relationship between the source token and the target token, where the relationship between the source token and the target token is one-to-many.
+
+The relationship between the target token and the source token is one-to-many, and the relationship is updated when a `link/unlink/updateTarget` operation occurs.
+
+**Root Token:**
+
+Each ERC-721 token linked to other ERC-721 token has a root token, and the owner of the root token is the owner of the entire graph. For a ERC-721 token, it's root token can be found through `findRootToken`.
+
+Inside `findRootToken` is done by constantly searching for a target token. When a token does not have a target token but has source tokens, it is a root token.
+
+**ERC-20/ERC-1155:**
+
+For ERC-20 token/ERC-1155 token, both have fungible attributes,  so need to link them to a ERC-721Token.
+
+**ERC-777 and other fungible tokens:**
+
+While most tokens are using ERC-20, there are some tokens that use ERC-777 or other fungible token standard, for these niche tokens you can use function names like `linkERC777/unlinkERC777/updateERC777Target`
+
+**Compose ERC-4786 with ERC-721 contracts:**
+
+If the contract code for ERC-4786 and ERC-721 is placed directly in one contract, this may cause the contract to exceed the size limit, so it is better to deploy it in two separate contracts. If you want to use a concept similar to the container NFT token, you can limit the targetToken of the link to only the NFTs minted in a particular contract  and check the target token of the updateTarget.
+
+## Backwards Compatibility
+The ERC-4786 is an extension of the ERC-721 and does not modify the interface of ERC-721, so it is 100% compatible with the ERC-721.
+
+## References
+
+**Standards**
+
+- [ERC-721 Non-Fungible Token Standard](https://eips.ethereum.org/EIPS/eip-721) 
+
+- [ERC-20 Token Standard](https://eips.ethereum.org/EIPS/eip-20)
+- [ERC-1155 Multi Token Standard](https://eips.ethereum.org/EIPS/eip-1155)
+- [ERC-165 Standard Interface Detection](https://eips.ethereum.org/EIPS/eip-165)
+- [ERC-998 Composable Non-Fungible Token Standard](https://eips.ethereum.org/EIPS/eip-998)
+
+**Implementations**
+
+- [Poria Cattus ERC-4786 reference implementation](https://github.com/poria-cat/ERC-4786)
+
+## Security Consideration
+There are no security considerations related directly to the implementation of this standard.
+
+## Citation
+
+Please cite this document as:
+
+[Poria Cattus](https://github.com/poria-cat), "EIP-4786: Link Common Token to ERC-721," *Ethereum Improvement Proposals*, no. 4786, February 2022. [Online serial]. Available: https://eips.ethereum.org/EIPS/eip-4786.
```

Comment:
```suggestion
## Copyright
Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).
```
Please see https://raw.githubusercontent.com/ethereum/EIPs/master/eip-template.md for an example of the proper layout of an EIP.

Reply:
Thanks @MicahZoltu , I get it. I also changed the Reference Implementation section.